### ISSUE TITLE
TensorFlowModel is not available in SageMaker Pipelines

### ISSUE THREAD

**Issue by matsuyusu** [aws/sagemaker-python-sdk#2791]
**Describe the bug**

For TensorFlowModel in sagemaker.tensorflow.model, if model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts, then we get an error. if model_data=s3/path/model.tar.gz, then the execution will be successful.

**To reproduce**
A clear, step-by-step set of instructions to reproduce the bug.

To use SageMaker Pipelines, we write pipeline.py. 
In pipeline.py, we must create a model to define a batch inference as follow.
```
from sagemaker.tensorflow.model import TensorFlowModel
model = TensorFlowModel(
    entry_point=os.path.join(BASE_DIR, 'inference.py'),
    model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts,
    source_dir=BASE_DIR,
    role=role,
    name='XXX',
    framework_version=TF_VERSION,
    sagemaker_session=sagemaker_session)
```

**Expected behavior**

Transform using a model created in TrainingStep.

**Screenshots or logs**
If applicable, add screenshots or logs to help explain your problem.

We get the following exception:
Exception: 'Properties' object has no attribute 'lower'

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**:2.70.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**: TensorFlow
- **Framework version**:1.15.2
- **Python version**:3.7
- **CPU or GPU**:CPU
- **Custom Docker image (Y/N)**:N

**Additional context**

**Comment by asmaier** [comment]
Is there a workaround for this issue? Hardcoding the model_data path doesn't help much, because the path chances every time the pipeline is running.

**Comment by matsuyusu** [comment]
We can use FrameworkModel or Model, which is bases of TensorFlowModel, as an alternative. However, it is not possible to embed inference.py and requirements.txt into model.tar.gz, so we need to deal with them separately.

**Comment by asmaier** [comment]
In my case even hardcoding the model_data path like

    model_data = "s3://test/model.tar.gz"
    
does not work. I get the same error: `Exception: 'Properties' object has no attribute 'lower'`. I tried to debug the issue and believe that in https://github.com/aws/sagemaker-python-sdk/blob/b02676922b76e23b2a46cb770eaa8bfb1076517d/src/sagemaker/workflow/step_collections.py#L201 the `model_data` string gets overwritten by some `Properties` leading to the exception. I tried to comment out that line, but then I got another confusing error `botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found` . 

So maybe there is more than one issue to be fixed here. But I don't understand the code good enough to really pinpoint the issue.

**Comment by matsuyusu** [comment]
![image](https://user-images.githubusercontent.com/95735278/148690707-8210aebb-ffbf-4669-9d4a-b09499a69630.png)
In my environment, the above code looks like the attached file, but it works. 
My code is as follows.
```
    model = TensorFlowModel(
        entry_point=os.path.join(BASE_DIR, 'inference.py'),
        model_data="s3:/test/model.tar.gz",
        source_dir=BASE_DIR,
        role=role,
        name='test',
        framework_version=TF_VERSION,
        sagemaker_session=sagemaker_session)
```

**Comment by qidewenwhen** [comment]
@matsuyusu, sorry for the inconvenience. 
The [fix](https://github.com/aws/sagemaker-python-sdk/commit/6ffe9258cd06cbbcdd46010afa724f31208ceec7) has been finally released in SM PySDK v2.85.0. Please check if it works for you. 

Closing this issue. Feel free to reopen if you have further questions.
